akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala

```diff
@@ -279,18 +285,25 @@ private[akka] final class ReplayingEvents[C, E, S](
       if (state.receivedPoisonPill && isInternalStashEmpty && !isUnstashAllInProgress)
         Behaviors.stopped
       else {
-        val running =
-          Running[C, E, S](
-            setup,
-            Running.RunningState[S](
-              seqNr = state.seqNr,
-              state = state.state,
-              receivedPoisonPill = state.receivedPoisonPill,
-              state.version,
-              seenPerReplica = state.seenSeqNrPerReplica,
-              replicationControl = Map.empty))
-
-        tryUnstashOne(running)
+        val runningState = Running.RunningState[S](
+          seqNr = state.seqNr,
+          state = state.state,
+          receivedPoisonPill = state.receivedPoisonPill,
+          state.version,
+          seenPerReplica = state.seenSeqNrPerReplica,
+          replicationControl = Map.empty)
+        val running = new Running(setup.setMdcPhase(PersistenceMdc.RunningCmds))
+        val initialRunningState = setup.replication match {
+          case Some(replication) => startReplicationStream(setup, runningState, replication)
+          case None              => runningState
+        }
+        setup.retention match {
+          case criteria: SnapshotCountRetentionCriteriaImpl if criteria.snapshotEveryNEvents <= state.eventsRecovered =>
+            internalSaveSnapshot(initialRunningState)
+            new running.StoringSnapshot(initialRunningState, immutable.Seq.empty, SnapshotWithoutRetention)
```

Comment:
Ok, and when snapshot storage completes (both if successfull and if failed) it also calls `tryUnstashOne`, so that anything that came in during recovery is handled. Good.